godot_fmod.cpp

```diff
@@ -578,109 +613,76 @@ FMOD_3D_ATTRIBUTES Fmod::get3DAttributes(FMOD_VECTOR pos, FMOD_VECTOR up, FMOD_V
 }
 
 void Fmod::playOneShot(const String &eventName, Object *gameObj) {
-	if (!eventDescriptions.has(eventName)) {
-		FMOD::Studio::EventDescription *desc = nullptr;
-		checkErrors(system->getEvent(eventName.ascii().get_data(), &desc));
-		eventDescriptions.insert(eventName, desc);
-	}
-	auto desc = eventDescriptions.find(eventName);
-	FMOD::Studio::EventInstance *instance;
-	checkErrors(desc->value()->createInstance(&instance));
-	if (instance) {
-		// set 3D attributes once
-		if (!isNull(gameObj)) {
-			updateInstance3DAttributes(instance, gameObj);
-		}
-		checkErrors(instance->start());
-		oneShotInstances.push_back(instance);
-	}
+	FMOD::Studio::EventInstance *instance = createInstance(eventName, true, false, nullptr);
+    if (instance) {
+        // set 3D attributes once
+        if (!isNull(gameObj)) {
+            updateInstance3DAttributes(instance, gameObj);
+        }
+        checkErrors(instance->start());
```

Comment:
You can issue a `checkErrors(instance->release());` immediately after starting it. Refer to my first comment on the `update()` method.